fastcrypto-zkp/src/conversions.rs

```diff
@@ -143,8 +143,18 @@ pub fn bls_g1_affine_to_blst_g1_affine(pt: &BlsG1Affine) -> blst_p1_affine {
     unsafe {
         blst_p1_affine_serialize(tmp2.as_mut_ptr(), &tmp_p1);
     };
-    let mut g1 = blst_p1_affine::default();
+    // BLST disagrees with Arkworks on the uncompressed representation of the infinity point on G1
+    // https://github.com/supranational/blst/blob/6382d67c72119d563975892ed49ba32e92d3d0da/src/e1.c#L153-L162
+    // the infinity bit notwithstanding,
+    // Arkworks: x = Fq::zero(), y = Fq::one()
+    // BLST: x = y = Fq::zero()
+    // BLST follows the standard here (see https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-pairing-friendly-curves-10#appendix-C)
+    if pt.infinity {
+        tmp2[0] = 0x40; // set the infinity bit
+        tmp2[48..].copy_from_slice(&[0u8; 48]); // zeroize the rest of the point
```

Comment:
Ok now I get it, but I guess for converting y=Fq::one() ->  Fq::zero() you only needed to set one bit Vs copy the whole 48 bytes slice?

Reply:
That works, if I'm being given a legitimate Arkworks point produced by the Arkworks library. 

Except Arkworks conveys the infinity using a separate flag here, so by using custom code, or simply abusing the somewhat permissive deserialization, the caller could send whatever they want in the second coordinate.

Reply:
interesting, so it's malleable, pretty bad IMO

Reply:
I'm happy to err on the safe side, so your approach meets the bar.